mailpoet/lib/AutomaticEmails/WooCommerce/Events/PurchasedInCategory.php

```diff
@@ -167,12 +174,14 @@ public function metaModifier(Newsletter $automaticEmail, array $meta): array {
     return $meta;
   }
 
-  private function getProductCategoryIdsMatchingNewsletterTrigger(Newsletter $automaticEmail, array $orderedCategoryIds): array {
-    $automaticEmailMeta = $automaticEmail->getMeta();
-    if (empty($automaticEmailMeta['option'])) {
+  private function getProductCategoryIdsMatchingNewsletterTrigger(NewsletterEntity $automaticEmail, array $orderedCategoryIds): array {
+    $automaticEmailMetaValue = $automaticEmail->getOptionValue(NewsletterOptionFieldEntity::NAME_META);
+    $optionValue = (is_string($automaticEmailMetaValue) && Helpers::isJson($automaticEmailMetaValue)) ? json_decode($automaticEmailMetaValue, true) : $automaticEmailMetaValue;
```

Comment:
The `is_string` check is unnecessary because `Helpers::isJson` function performs it too.

Reply:
Good point. I thought by the name of the parameter in Helpers::isJson() that it accepts string only. Thank you